.github/workflows/package_for_test.yml

```diff
@@ -135,8 +162,29 @@ jobs:
           name: Neuron-Linux
           path: release/Neuron-*.AppImage
 
+  cache-package-info:
+    needs: [packaging]
+    if: ${{ github.event_name == 'pull_request' }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Save PR number and run id
+        env:
+          PR_NUMBER: ${{ github.event.number }}
+          RUN_ID: ${{ github.run_id }}
+          SHA: ${{ github.event.pull_request.head.sha }}
+        run: |
+          mkdir -p ./package_info
+          echo $PR_NUMBER > ./package_info/pr_number
+          echo $RUN_ID > ./package_info/run_id
+          echo $SHA > ./package_info/sha
+      - uses: actions/upload-artifact@v3
+        with:
+          name: package_info
+          path: package_info/
+          retention-days: 3
   comment:
     needs: [packaging]
+    if: ${{ github.event_name == 'push' }}
```

Comment:
I think we should add comments to these places where the background reasons are not easily understood from the code, to avoid them becoming increasingly difficult to maintain.